File: Arrays/01-prefix-sum-i.c
#include<stdio.h>

void prefix(int nums[], int n) {

    for(int i = 1; i < n; i++) {
        nums[i] += nums[i-1];
    }
}

void printArray(int nums[], int n) {

    printf("Array Elements are: \n");
    for(int i = 0; i < n; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
}

int main() {
    int nums[] = {1,2,4,3,2};
    int n = sizeof(nums) / sizeof(int);
    printArray(nums, n);
    prefix(nums, n);
    printArray(nums, n);
    return 0;
}

File: Arrays/01-prefix-sum-i.js
const prefix = (nums) => {
    for(let i = 1; i < nums.length; i++) {
        nums[i] += nums[i - 1];
    }
    return nums;
}

function main() {
    let nums = [1,2,4,3,5];
    console.log(nums);
    nums = prefix(nums);
    console.log(nums);
}

main();